src/attack.c

```diff
@@ -591,15 +593,24 @@ check_host_authorization (gvm_host_t *host, const struct in6_addr *addr,
   hosts_deny = gvm_hosts_new (prefs_get ("hosts_deny"));
   if (!host_authorized (host, addr, hosts_allow, hosts_deny))
     {
-      error_message_to_client2 (kb, "Host access denied.", NULL);
+      char ip_str[INET6_ADDRSTRLEN];
+
+      addr6_to_str (addr, ip_str);
+      error_message_to_client2 (kb, "Host access denied.", ip_str, NULL);
```

Comment:
We could let the caller handle the error reporting. We already print a warning in `attack_start` if access denied. We could send the message there so that `check_host_authorization` is only responsible for checking and nothing more. This way we do not need to do `addr6_to_str` two times.